x-pack/plugins/security_solution/public/timelines/components/timeline/helpers.tsx

```diff
@@ -263,7 +262,7 @@ export const buildIsQueryMatch = ({
   } else if (checkIfFieldTypeIsDate(field, browserFields)) {
     return convertDateFieldToQuery(field, value);
   } else {
-    return `${field} : ${isNumber(value) ? value : escapeQueryValue(value)}`;
+    return `${field} : ${prepareKQLParam(value)}`;
   }
```

Comment:
I have one question here @maximpn & @banderror  , Previously, if the value was number, it was being passed as number but with the new changes, I see below line in `prepareKQLParam`. Here, we are affecting everything to a string. I can see tests passing fine but does it work fine on number fields where KQL is searching them as `strings` ?

``` javascript
  return isString(value) ? prepareKQLStringParam(value) : `${value}`;
```

Please let me know if I missed something.